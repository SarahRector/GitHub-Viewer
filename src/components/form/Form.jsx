import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchProfile, fetchRepos, setLookup } from '../../actions/repoActions';

const Form = () => {
  const dispatch = useDispatch();
  const userName = useSelector(state => state.lookup);

  const handleChange = ({ target }) => {
    dispatch(setLookup(target.value));
  };

  const handleSubmit = () => {
    dispatch(fetchProfile(userName));
    dispatch(fetchRepos(userName));
  };

  return (
    <>
      <label htmlFor="userLookup">User Lookup</label>
      <input
        id="userLookup"
        type="text"
        name="userLookup" 
        onChange={handleChange} />
      <button onClick={handleSubmit}>Search!</button>
    </>
  );
};

export default Form;
