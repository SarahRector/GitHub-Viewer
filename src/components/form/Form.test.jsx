import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Form from './Form';
import { Provider } from 'react-redux';
import store from '../../store';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders Form', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Form />
      </Provider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
