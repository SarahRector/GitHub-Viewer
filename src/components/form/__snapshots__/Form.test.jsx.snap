// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders Form 1`] = `
<DocumentFragment>
  <label
    for="userLookup"
  >
    User Lookup
  </label>
  <input
    id="userLookup"
    name="userLookup"
    type="text"
  />
  <button>
    Search!
  </button>
</DocumentFragment>
`;
